core-ui/src/resources/CustomResourceDefinitions/ExtensibilityStarterModal/helpers.js

```diff
@@ -0,0 +1,142 @@
+    .filter(filterSimpleProps)
+    .map(([name, property]) => ({
+      path: `spec.${name}`,
+      type: property.type,
+      name,
+      isSelected: true,
+      required: requiredProperties.includes(name),
+    }));
+}
+
+function extractAdditionalPrinterColumns(crd) {
+  const version = crd.spec.versions.find(v => v.storage);
+
+  return (version.additionalPrinterColumns || []).map(apc => ({
+    path: apc.jsonPath.substring(1), // remove trailing '.'
+    type: apc.type,
+    name: apc.name,
+    isSelected: true,
+  }));
+}
+
+function isStatusMaybe({ name }) {
+  return name.toLowerCase() === 'status' || name.toLowerCase() === 'phase';
+}
+
+export function extractValueColumns(crd) {
+  const firstLevelPropertiesColumns = extractFirstLevelProperties(crd);
+  const additionalPrinterColumns = extractAdditionalPrinterColumns(crd);
+
+  const inferWidgets = c => ({
+    ...c,
+    ...(isStatusMaybe(c) ? { widget: 'Badge' } : {}),
+  });
+
+  return [...firstLevelPropertiesColumns, ...additionalPrinterColumns].map(
+    inferWidgets,
+  );
+}
+
+export function createExtensibilityTemplate(crd, t) {
+  const version = crd.spec.versions.find(v => v.storage);
+  const additionalValueColumns = extractValueColumns(crd);
+  const possibleStatusColumn = additionalValueColumns.find(isStatusMaybe);
+
+  return {
+    resource: {
+      kind: crd.spec.names.kind,
+      group: crd.spec.group,
+      version: version.name,
+    },
+    navigation: {
+      path: crd.spec.names.plural,
+      scope: crd.spec.scope === 'Namespaced' ? 'namespace' : 'cluster',
+    },
+    form: extractFirstLevelProperties(crd),
+    list: additionalValueColumns,
+    details: {
+      header: possibleStatusColumn ? [possibleStatusColumn] : [],
+      body: [
+        {
+          name: 'Summary',
+          widget: 'Panel',
+          children: additionalValueColumns.length
+            ? extractValueColumns(crd) // create a copy
+            : [
+                {
+                  name: 'Created at',
+                  isSelected: true,
+                  path: 'metadata.creationTimestamp',
+                  type: 'date',
+                },
+              ],
+        },
+        { path: 'spec', widget: 'CodeViewer' },
+      ],
+    },
+    // turns out the openAPIV3Schema is not required
+    schema: version.schema?.openAPIV3Schema || {},
+    translations: {
+      en: {
+        name: pluralize(prettifyKind(crd.spec.names.kind)),
+        category: 'Custom Resources',
+        ...(additionalValueColumns.length
+          ? { 'metadata.creationTimestamp': 'Created at' }
+          : {}),
+      },
+    },
+  };
+}
+
+export function createConfigmap(crd, data) {
+  const filterViewProps = arr =>
+    arr
+      .filter(e => e.isSelected)
+      .map(e => {
+        delete e.isSelected;
+        delete e.required;
+        return e;
+      });
+
+  data.list = filterViewProps(data.list);
+  data.details.body[0].children = filterViewProps(
```

Comment:
Maybe we should remove `panel` if we don't have any children

Reply:
We would always have some children - either squeezed out of schema/`additionalPrinterColumns` or single `Created at` column as a fallback.

Reply:
If we un check all the checkboxes on creation form then we have empty panel